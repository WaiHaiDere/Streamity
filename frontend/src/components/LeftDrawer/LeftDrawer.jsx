import React from "react";
import { Drawer, List, IconButton } from "@material-ui/core";
import PropTypes from "prop-types";

import SearchIcon from "@material-ui/icons/Search";
import Input from "@material-ui/core/Input";
import InputLabel from "@material-ui/core/InputLabel";
import InputAdornment from "@material-ui/core/InputAdornment";
import FormControl from "@material-ui/core/FormControl";
import SearchResult from "../SearchResult/SearchResult";
import styles from "./leftdrawer.module.css";
import AddIcon from "@material-ui/icons/Add";

const LeftDrawer = ({
  listOfSearchResults,
  handleChange,
  handleClickSearch,
  addToPlaylist,
}) => {
  return (
    <div>
      <Drawer
        variant="permanent"
        anchor="left"
        classes={{ paper: styles.Drawer }} // Need to use Paper here instead of root for the drawer.
      >
        <div className={styles.SearchBox}>
          <FormControl>
            <InputLabel
              htmlFor="search-input"
              classes={{
                root: styles.searchColour,
              }}
            >
              Search
            </InputLabel>
            <Input
              id="search-input"
              onChange={handleChange}
              endAdornment={
                <InputAdornment position="end">
                  <IconButton
                    aria-label="send search"
                    classes={{
                      root: styles.searchColour,
                    }}
                    onClick={handleClickSearch}
                  >
                    <SearchIcon />
                  </IconButton>
                </InputAdornment>
              }
            />
          </FormControl>
        </div>
        <List>
          {listOfSearchResults.map((searchResult) => {
            return (
              <>
              <div className={styles.SearchResult}>
                <SearchResult
                  key={searchResult.trackUri}
                  title={searchResult.songName}
                  artist={searchResult.artist[0].name}
                  albumArt={searchResult.album.images[0].url}
                />
                <IconButton
                  key={`${searchResult.trackUri}-add`}
                  onClick={() => {
                    addToPlaylist(searchResult);
                  }}
                >
                  <AddIcon />
                </IconButton>
                </div>
              </>
            );
          })}
        </List>
      </Drawer>
    </div>
  );
};

LeftDrawer.defaultProps = {
  listOfSearchResults: [{}],
  handleChange: () => {},
  handleClickSearch: () => {},
  addToPlaylist: () => {},
};

LeftDrawer.propTypes = {
  listOfSearchResults: PropTypes.arrayOf(PropTypes.object),
  handleChange: PropTypes.func,
  handleClickSearch: PropTypes.func,
  addToPlaylist: PropTypes.func,
};

export default LeftDrawer;
