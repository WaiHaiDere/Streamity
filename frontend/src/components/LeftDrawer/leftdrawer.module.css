.Drawer {
  width: 280px !important;
}

.SearchBox {
  display: flex !important;
  justify-content: center !important;
  padding-top: 10px;
  padding-bottom: 20px;
  background-color: #527cec;
}

.SearchResultContainer {
  display: flex;
  align-items: center;
  align-content: center;
  padding: 10px;
}

.SearchResult {
  width: 70%;
}

.searchColour {
  color: white;
}
