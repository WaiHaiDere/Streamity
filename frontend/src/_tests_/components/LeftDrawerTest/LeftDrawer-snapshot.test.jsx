import React from "react";
import renderer from "react-test-renderer";

import LeftDrawer from "../../../components/LeftDrawer/LeftDrawer";

describe("Test left drawer snapshot", () => {
  it("left drawer renders correctly", () => {
    const mockOnClick = jest.fn();
    const mockSearchResults = [
      {
        album: {
          name: "SIX60",
          images: [{ url: "hello.jpg" }],
        },
        artist: "SIX60",
        duration: "0",
        songName: "Catching Feelings",
        trackUri: "12345",
      },
    ];

    const tree = renderer
      .create(
        <LeftDrawer
          listOfSearchResults={mockSearchResults}
          handleChange={mockOnClick}
          handleClickSearch={mockOnClick}
          addToPlaylist={mockOnClick}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
