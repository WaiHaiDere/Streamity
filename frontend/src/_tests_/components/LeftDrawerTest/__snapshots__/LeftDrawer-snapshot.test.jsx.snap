// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test left drawer snapshot left drawer renders correctly 1`] = `
<div>
  <div
    className="MuiDrawer-root MuiDrawer-docked"
  >
    <div
      className="MuiPaper-root MuiDrawer-paper Drawer MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft MuiPaper-elevation0"
    >
      <div
        className="SearchBox"
      >
        <div
          className="MuiFormControl-root"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root searchColour MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink={false}
            htmlFor="search-input"
          >
            Search
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              className="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
              disabled={false}
              id="search-input"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
            />
            <div
              className="MuiInputAdornment-root MuiInputAdornment-positionEnd"
            >
              <button
                aria-label="send search"
                className="MuiButtonBase-root MuiIconButton-root searchColour"
                disabled={false}
                onBlur={[Function]}
                onClick={[MockFunction]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                    />
                  </svg>
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
      <ul
        className="MuiList-root MuiList-padding"
      >
        <div
          className="SearchResult"
        >
          <li
            className="MuiListItem-root alignItems MuiListItem-gutters"
            disabled={false}
          >
            <img
              alt="album-art"
              className="searchResultIcon"
              src="hello.jpg"
            />
            <div
              className="MuiListItemText-root listItem MuiListItemText-multiline"
            >
              <span
                className="MuiTypography-root MuiListItemText-primary itemPrimaryText MuiTypography-body1 MuiTypography-displayBlock"
              >
                Catching Feelings
              </span>
              <p
                className="MuiTypography-root MuiListItemText-secondary itemSecondaryText MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
              >
                
              </p>
            </div>
          </li>
          <button
            className="MuiButtonBase-root MuiIconButton-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                />
              </svg>
            </span>
          </button>
        </div>
      </ul>
    </div>
  </div>
</div>
`;
